torch/lobpcg.py

```diff
@@ -0,0 +1,214 @@
+"""Locally Optimal Block Preconditioned Conjugate Gradient methods
+
+
+References:
+
+  [1] Andrew V. Knyazev. (2001) Toward the Optimal Preconditioned Eigensolver:
+  Locally Optimal Block Preconditioned Conjugate Gradient Method. SIAM
+  J. Sci. Comput., 23(2), 517–541. (25 pages)
+  https://epubs.siam.org/doi/abs/10.1137/S1064827500366124
+  A report version: http://ccm.ucdenver.edu/reports/rep149.pdf
+
+  [2] Jed A. Duersch, Meiyue Shao, Chao Yang, Ming Gu. (2018)  A Robust and
+  Efficient Implementation of LOBPCG.  SIAM J. Sci. Comput., 40(5),
+  C655–C676. (22 pages)
+  https://epubs.siam.org/doi/abs/10.1137/17M1129830
+  https://arxiv.org/pdf/1704.07458.pdf
+
+"""
+
+import torch
+
+from .lowrank import get_matmul, batch_transpose, get_floating_dtype
+
+
+def RayleighRitz(S, A, B=None, out=None):
+    """Rayleigh-Ritz procedure - solve projected eigenvalue problem
+
+    .. note:: The implementation is based on the Algorithm 2 from
+              Duersch et al, 2018.
```

Comment:
The article name for `Duersch et al` seems to be mentioned only at the top of lobpcg.py and won't be visible in public html docs. I think it'd be nice to include it to `lobpcg_basic` and maybe `RayleighRitz` docstrs

Reply:
Moved the references to the functions documentation strings.